config/src/main/java/com/typesafe/config/impl/SimpleConfigObject.java

```diff
@@ -596,7 +597,12 @@ public boolean equals(Object other) {
     public int hashCode() {
         // note that "origin" is deliberately NOT part of equality
         // neither are other "extras" like ignoresFallbacks or resolve status.
-        return mapHash(this);
+        if (this.mapHashValue == null) {
+            return this.mapHashValue=mapHash(this);
+        }
```

Comment:
is there some theoretical thread-safety issue here, like the mapHashValue being available to another thread before the Integer is fully constructed? I don't know my java concurrency well enough to say - @ktoso any thoughts?

minor style thing, I think the code would mostly have spaces around that `=` and pull the following `else` up next to the above brace.

Other than those things this looks good, thanks!

Reply:
Thanks for the style suggestions.
About the thread-safety, there is no issue due to the immutability of the structures of SimpleConfigObject. So, I avoided any synchronization. In the worst case, the `mapHash` will be called more than once in parallel and the value of `mapHashValue` will be written more than once, but with the same value. It'll be no issue, and will be much faster than any other synchronization approach.